go/cmd/dolt/commands/sqlserver/sqlserver.go

```diff
@@ -289,5 +297,50 @@ func getYAMLServerConfig(fs filesys.Filesys, path string) (ServerConfig, error)
 		return nil, fmt.Errorf("Failed to parse yaml file '%s'. Error: %s", path, err.Error())
 	}
 
+	if cfg.PersistenceBehavior() == loadPerisistentGlobals {
```

Comment:
these changes look correct, but it might be simpler to structure the code in a way that naturally describes what's happening here. For every variable, 1) we try to set it from command line arg, 2) we try to set it from persisted globals, 3) we resort to a default value. The output config is considered final, because this function has taken care of all possible inputs and applied the correct ordering (? I don't actually know if it's true, but if not it would make sense to add those here, right?)

Reply:
if it is possible to structure the function like above I think it makes sense to add a table-test that runs `getCommandLineServerConfig` with different sets of arg and config inputs. You can test every variable rather than getting one case in a bats test.

Reply:
I wasn't able to make tests explicitly for `getCommandLineServerConfig`, but I added more bats test.